hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanActionExecutor.java

```diff
@@ -159,9 +162,19 @@ protected Option<HoodieCleanerPlan> requestClean(String startCleanTime) {
         LOG.error("Got exception when saving cleaner requested file", e);
         throw new HoodieIOException(e.getMessage(), e);
       }
-      return Option.of(cleanerPlan);
+      option = Option.of(cleanerPlan);
     }
-    return Option.empty();
+
+    if (config.getTableManagerConfig().isTableManagerSupportsAction(ActionType.clean)) {
+      delegateCleanExecutionToTableManager();
+    }
```

Comment:
this reads like "as long as table manager supports this action, we delegate it". but internally it checks enabling or not. we should avoid this confusion.

We should also make the logic flow cleanly separated: can you make the delegation return Option<HoodieCleanerPlan> and fully separated from the existing flow? if to delegate, then let the delegate do everything. This applies to other executor changes.